mlflow/openai/__init__.py

```diff
@@ -698,91 +668,102 @@ def get_params_list(self, data):
+            from openai import AzureOpenAI
 
-            return (
-                f"{api_base}/openai/deployments/{deployment_id}/"
-                f"{task_url}?api-version={api_version}"
+            return AzureOpenAI(
+                api_key=self.api_token.token,
+                azure_endpoint=self.api_config.api_base,
+                api_version=self.api_config.api_version,
+                azure_deployment=self.api_config.deployment_id,
             )
+        else:
+            from openai import OpenAI
 
-        return f"{api_base}/{task_url}" if api_base else default_url
+            return OpenAI(
+                api_key=self.api_token.token,
+                base_url=self.api_config.api_base,
+            )
 
     def _predict_chat(self, data, params):
         from mlflow.openai.api_request_parallel_processor import process_api_requests
 
         _validate_model_params(self.task, self.model, params)
+        max_retries = params.pop("max_retries", self.api_config.max_retries)
+        timeout = params.pop("timeout", self.api_config.timeout)
+
         messages_list = self.format_completions(self.get_params_list(data))
-        requests = [{**self.model, **params, "messages": messages} for messages in messages_list]
-        request_url = self._construct_request_url("chat/completions", REQUEST_URL_CHAT)
-
-        results = process_api_requests(
-            requests,
-            request_url,
-            api_token=self.api_token,
-            max_requests_per_minute=self.api_config.max_requests_per_minute,
-            max_tokens_per_minute=self.api_config.max_tokens_per_minute,
-        )
-        return [r["choices"][0]["message"]["content"] for r in results]
+        client = self.client.with_options(max_retries=max_retries, timeout=timeout)
+
+        requests = [
+            partial(
+                client.chat.completions.create,
+                messages=messages,
+                model=self.model["model"],
+                **params,
+            )
+            for messages in messages_list
+        ]
+
+        results = process_api_requests(requests=requests)
+
+        return [r.choices[0].message.content for r in results]
 
     def _predict_completions(self, data, params):
         from mlflow.openai.api_request_parallel_processor import process_api_requests
 
         _validate_model_params(self.task, self.model, params)
         prompts_list = self.format_completions(self.get_params_list(data))
-
+        max_retries = params.pop("max_retries", self.api_config.max_retries)
+        timeout = params.pop("timeout", self.api_config.timeout)
         batch_size = params.pop("batch_size", self.api_config.batch_size)
         _logger.debug(f"Requests are being batched by {batch_size} samples.")
+
+        client = self.client.with_options(max_retries=max_retries, timeout=timeout)
+
         requests = [
-            {
-                **self.model,
+            partial(
+                client.completions.create,
```

Comment:
Not related to this PR: OpenAI deprecated completion API a while ago, do you think it makes sense to drop support in MLflow as well? cc: @BenWilson2 @harupy

Reply:
I think we can remove it once OpenAI removes it: https://community.openai.com/t/completion-models-are-now-considered-legacy/656302/3 implies it's marked as legacy but not deprecated yet.

Reply:
I'm curious if other LLM packages still support it or not.

Reply:
Yea aligns with OpenAI's status makes sense

Reply:
Most integration packages have dropped it, but we still have stuff on our runtime side that supports it (legacy endpoints). We'll wait for everything to be cleaned up and removed before dropping support :)